binding/react/src/use_leopard.ts

```diff
@@ -0,0 +1,254 @@
+    onmessage: (event: any) => {
+      switch (event.data.command) {
+        case 'process':
+          audioDataRef.current.push(event.data.inputFrame);
+          break;
+        default:
+          break;
+      }
+    },
+  });
+
+  const init = useCallback(
+    async (
+      accessKey: string,
+      model: LeopardModel,
+      options: LeopardOptions = {},
+      processOptions: {
+        transfer?: boolean;
+        transferCallback?: (data: Int16Array) => void;
+      } = {}
+    ): Promise<void> => {
+      try {
+        if (!leopardRef.current) {
+          leopardRef.current = await LeopardWorker.create(
+            accessKey,
+            model,
+            options
+          );
+
+          processOptionsRef.current = processOptions;
+          setIsLoaded(true);
+          setError(null);
+        }
+      } catch (e: any) {
+        setError(e);
+      }
+    },
+    []
+  );
+
+  const process = useCallback(async (pcm: Int16Array): Promise<void> => {
+    try {
+      if (!leopardRef.current) {
+        setError(
+          new Error('Leopard has not been initialized or has been released')
+        );
+        return;
+      }
+
+      const processResult = await leopardRef.current.process(
+        pcm,
+        processOptionsRef.current
+      );
+      setResult(processResult);
+    } catch (e: any) {
+      setError(e);
+    }
+  }, []);
+
+  const processFile = useCallback(async (audioFile: File): Promise<void> => {
+    if (!leopardRef.current) {
+      setError(
+        new Error('Leopard has not been initialized or has been released')
+      );
+      return new Promise(resolve => resolve());
+    }
+
+    // @ts-ignore
+    const audioContext = new (window.AudioContext || window.webKitAudioContext)(
+      { sampleRate: leopardRef.current.sampleRate }
+    );
+
+    const readAudioFile = (
+      file: File,
+      callback: (audioBuffer: AudioBuffer) => Promise<void>
+    ): void => {
+      const reader = new FileReader();
+
+      reader.onload = (): void => {
+        const wavBytes = reader.result;
+        audioContext.decodeAudioData(wavBytes as ArrayBuffer, callback);
+      };
+
+      reader.readAsArrayBuffer(file);
+    };
+
+    return new Promise<void>(resolve => {
+      readAudioFile(audioFile, async (audioBuffer: AudioBuffer) => {
+        const f32PCM = audioBuffer.getChannelData(0);
+        const i16PCM = new Int16Array(f32PCM.length);
+        const INT16_MAX = 32767;
+        const INT16_MIN = -32768;
+        i16PCM.set(
+          f32PCM.map(f => {
+            let i = Math.trunc(f * INT16_MAX);
+            if (f > INT16_MAX) i = INT16_MAX;
+            if (f < INT16_MIN) i = INT16_MIN;
+            return i;
+          })
+        );
+
+        await process(i16PCM);
+        resolve();
+      });
+    });
+  }, []);
+
+  const stopRecording = useCallback(async (): Promise<void> => {
+    try {
+      if (timerRef.current) {
+        clearInterval(timerRef.current);
+
```

Comment:
was this meant to just wrap the `clearInterval` or was it intended to be the mark of whether we are recording or not? You do want to check the recording state, and if we're not recording, return without doing anything.